src/odemis/model/_futures.py

```diff
@@ -538,8 +546,20 @@ def _on_future_done(self, f):
             self.set_result(None)
 
     def _estimate_end(self):
-        start, end = self.get_progress()
-        return start + sum(self.futures[f] for f in self.futures if not f.done())
+        """
+        Estimate the end of the batch future based on the current time plus the
+        estimated time for all futures that still need to be executed.
+
+        returns: (float) Newly estimated end time based for the batch future.
+        """
+        duration = time.time()
+        for f, t in self.futures.items():
+            # only take futures into account that are not executed (finished) yet
+            if not f.done():
+                duration += t
+        # FIXME: it does not update often enough in the GUI, the estimate seems reasonable, but
+        #  gauge does not execute 1s when showing 1 second has passed -> takes longer thought time would match well
+        return duration
```

Comment:
It's supposed to return the time at which it expects to end. So that variable shouldn't be called `duration`. I think it'd be fine to write: `return time.time() + time_left` .